sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java

```diff
@@ -52,6 +54,7 @@
 
     private List<CredentialResult> adminKubeConfigs;
     private List<CredentialResult> userKubeConfigs;
+    private final Map<Format, List<CredentialResult>> formatUserKubeConfigsMap = new ConcurrentHashMap<>();
```

Comment:
Does it need to be cleaned in `refresh` (`getInnerAsync`)?

Reply:
Yes it should! Thanks for pointing out.